### ISSUE TITLE
CVA6 core failing to build

### ISSUE THREAD

**Issue by yadu-kr** [esa-tu-darmstadt/tapasco-riscv#32]
I'm trying to build the CVA6 core and generate the bitstream to load the PE onto an ZCU102 board. Unfortunately, the build process does not seem to terminate properly, and I'm faced with this error when I run `make cva6_pe`:

```
ERROR: [BD 41-1075] Cannot create master segment for slave segment </cva6_timer_0/axi_timer/reg0> in address space </cva6_0/io_axi_mem> at offset and range 0x1300_0000 [ 768K ]. Range = 0x000C_0000. Memory ranges must be a power of 2..
Exec Tcl: create_bd_segment -offset <0x13000000> -range <0x00000000000C0000> -slave_segment </cva6_timer_0/axi_timer/reg0> -parent_addr_space </cva6_0/io_axi_mem> -name <SEG_cva6_clint>
ERROR: [BD 5-14] Error: running create_bd_segment.
ERROR: [Common 17-39] 'create_bd_addr_seg' failed due to earlier errors.

    while executing
"create_bd_addr_seg -range $CLINT_LENGTH -offset $CLINT_BASE [get_bd_addr_spaces cva6_0/io_axi_mem] [get_bd_addr_segs cva6_timer_0/axi_timer/reg0] SEG_..."
    (procedure "create_specific_addr_segs" line 23)
    invoked from within
"create_specific_addr_segs"
    (file "common/common_addr_segments.tcl" line 46)

    while executing
"source common/common_addr_segments.tcl"
    (procedure "cr_bd_riscv_pe" line 57)
    invoked from within
"cr_bd_riscv_pe "" $lmem"
    (file "riscv_pe_project.tcl" line 159)
INFO: [Common 17-206] Exiting Vivado at Wed Oct  4 12:21:17 2023...
make: *** [Makefile:29: cva6_pe] Error 1
```

Could I get some help with this? Also, as I understand it, once the PE is generated, it is also imported automatically into Tapasco. What are the next steps to generate and load the core onto the FPGA? Thanks!

**Comment by cahz** [comment]
I unfortunately cannot reproduce the issue. My log output has the following:
```
WARNING: [BD 41-2721] Requested assignment from slave segment '/cva6_timer_0/axi_timer/reg0' to address space '/cva6_0/io_axi_mem' was auto-corrected from <0x1300_0000 [ 768K ]> to <0x1300_0000 [ 512K ]>.
```

Which version of Vivado do you use?

The next steps would be (see also the TaPaSCo readme):
* Build a bitstream: `tapasco compose [cva6_pe x 1] @ 100 MHz -p zcu102`
* Boot a linux image on the ZCU102 with interrupts enabled for TaPaSCo
* SSH to the ZCU102 and load the bitstream: `tapasco-load-bitstream <bit-file>.bit`
* Execute your host application (example available in tapasco-riscv/programming)

**Comment by yadu-kr** [comment]
Latest develop branch changes do not fix the problem with Vivado 2019.1, I still get the same error (running `create_bd_segment`) that I mentioned when I opened this issue.
As far as the log you require is concerned, that file is not generated with 2019.1 as the core build process fails before that, at the `make cva6_pe` step.

When using Vivado 2020.2, the core build process completes successfully, but I'm unfortunately not able to generate the bitstream using `tapasco compose` due to the missing part license.

**Comment by cahz** [comment]
I have pushed a change, which increases the address range

**Comment by yadu-kr** [comment]
The commit [99384ed](https://github.com/esa-tu-darmstadt/tapasco-riscv/commit/99384ed2de4ad6cbaabc4f93d44f6ea71bf88c51) has indeed fixed the problem with Vivado 2019.1, hence closing this issue. Thanks!
